File: src/main/java/rlbot/gamestate/BallState.java
package rlbot.gamestate;

import com.google.flatbuffers.FlatBufferBuilder;
import rlbot.flat.DesiredBallState;
import rlbot.flat.DesiredPhysics;

/**
 * See https://github.com/RLBot/RLBotJavaExample/wiki/Manipulating-Game-State
 */
public class BallState {
    private PhysicsState physics;

    public BallState() {
    }

    public BallState(PhysicsState physics) {
        this.physics = physics;
    }

    public PhysicsState getPhysics() {
        return physics;
    }

    public BallState withPhysics(PhysicsState physics) {
        this.physics = physics;
        return this;
    }

    public int toFlatbuffer(FlatBufferBuilder builder) {

        Integer physicsOffset = physics == null ? null : physics.toFlatbuffer(builder);

        DesiredBallState.startDesiredBallState(builder);
        if (physicsOffset != null) {
            DesiredBallState.addPhysics(builder, physicsOffset);
        }
        return DesiredPhysics.endDesiredPhysics(builder);
    }
}

File: src/main/cs/RLBotDotNet/RLBotDotNet/GameState/BallState.cs
using rlbot.flat;
using FlatBuffers;

namespace RLBotDotNet.GameState
{
    public class BallState
    {
        private PhysicsState physicsState;

        public PhysicsState PhysicsState
        {
            get
            {
                if (physicsState == null)
                    physicsState = new PhysicsState();

                return physicsState;
            }

            set => physicsState = value;
        }

        public BallState(PhysicsState physicsState = null)
        {
            PhysicsState = physicsState;
        }

        public BallState(BallInfo ballInfo)
        {
            if (ballInfo.Physics.HasValue)
                PhysicsState = new PhysicsState(ballInfo.Physics.Value);
        }

        public Offset<DesiredBallState> ToFlatBuffer(FlatBufferBuilder builder)
        {
            int physicsStateOffset = PhysicsState?.ToFlatBuffer(builder).Value ?? -1;

            DesiredBallState.StartDesiredBallState(builder);

            if (PhysicsState != null)
                DesiredBallState.AddPhysics(builder, new Offset<DesiredPhysics>(physicsStateOffset));

            return DesiredBallState.EndDesiredBallState(builder);
        }
    }
}